### PR TITLE
fixed modifcation parsing

### PR DESCRIPTION
**Pull request by csbrasnett**
- -mods flag now only read once for all modifications
- patch all molecule termini as proteins, so any modification in addition to these ones are also applied.
- add more modifications from vermouth to martini3 forcefield
- change logging level and fix tests as per #385

### CODE REVIEW

File: polyply/data/martini3/modifications.ff [hunk 1]
```diff
@@ -35,3 +35,72 @@ BB {"replace": {"atype": "P1", "charge": 0.0}}
...
+[ atoms ]
+BB  {"resname": "HIS", "replace": {"resname": "HIS"}}
+SC1 {"resname": "HIS", "replace": {"resname": "HIS"}}
+SC2 {"resname": "HIS", "replace": {"atype": "TP1dq", "charge": "0.5", "resname": "HIS"}}
+SC3 {"resname": "HIS", "replace": {"atype": "TP1dq", "charge": "0.5", "resname": "HIS"}}
+;[ edges ]
+;BB SC1
+;SC1 SC2
+;SC1 SC3
+;SC2 SC3
```
Comment (fgrunewald) [inline polyply/data/martini3/modifications.ff:80]: are these commented out on purpose?

File: polyply/src/apply_modifications.py [hunk 2]
```diff
@@ -77,12 +76,12 @@ def apply_mod(meta_molecule, modifications):
         target_residue = meta_molecule.nodes[target_resid - 1]
         # takes care to skip all residues that come from an itp file
         if not target_residue.get('from_itp', 'False'):
-            LOGGER.warning("meta_molecule has come from itp. Will not attempt to modify.")
+            LOGGER.info("meta_molecule has come from itp. Will not attempt to modify.")
```
Comment (fgrunewald) [inline polyply/src/apply_modifications.py:79]: having a test for this one would be ideal

**Review by fgrunewald — CHANGES_REQUESTED** [review]
Looks good but that one itp test case I would suggest to include; also could we have one test case where more than just one modification is added like terminal plus histidine?

File: polyply/data/martini3/modifications.ff (hunk 1 above)
Comment (csbrasnett) [reply polyply/data/martini3/modifications.ff:80]: yes. For some reason modifications.ff can't be read if they're included, and excluding them doesn't seem to affect them being applied correctly :man_shrugging:

**Comment by csbrasnett** [comment]
Re: the ignoring if coming from an itp test, two quick questions:

1) I can't tell why it's not picked up from https://github.com/marrink-lab/polyply_1.0/pull/389/files#diff-da096f96695964aa898c2ba48751a6ffa6465bd64f22753261d8536db11d5d24R98-R109 anyway?

2) I'm not sure nodes are ever given an attribute from 'from_itp' when they're read in via that meta_molecule subclass?

**Review by fgrunewald — APPROVED** [review]
looks good to me; @csbrasnett can we merge it?

**Comment by csbrasnett** [comment]
nothing more from me on this, go ahead and merge thanks! Can probably close #385 at the same time?

**Pull request merged**